resources/analysisTools/rnaseqworkflow/featureCountsDexseq_2_FpkmTpm

```diff
@@ -79,9 +79,18 @@ if (length $gtf_file > 0) {
     chomp $line;
     my @line_split = split ("\t", $line);
     if ($line_split[2] eq "gene") {
-      my @comment = split ("\"",$line_split[8]);
-      $gtf{$comment[1]}="$line_split[0]\t$line_split[3]\t$line_split[4]\t$comment[1]\t.\t$line_split[6]\t$comment[9]";
-      $gtf_names{$comment[1]}=$comment[9];
+      my @comment = split (";",$line_split[8]);
```

Comment:
If you touch the code, could you also make it more robust? Substitute the `my $line = <$gtf_fh>` stuff by 

```perl
while (! eof($gtf_fh)) {
   my $line = readline($gtf_fh) \
      || die "Error reading from '$gtf_file': $!";
   ...
}
```

If there is an I/O error on the reference-data mount (and they **do** occur occasionally!), the current Perl code just stops the loop with `$line == undef` but won't check for any I/O errors and silently continues! This means the GTF will only be partially processed, potentially invalidating the results.

Have a look at https://gitlab.com/one-touch-pipeline/otp/-/wikis/Workflow-Development/Coding-Guidelines.

Reply:
I didn't want to disturb the code base, as you said since we are almost there, I have added a few file handling as you have suggested.